### ISSUE TITLE
Setup local development environment for NetworkX

### ISSUE THREAD

**Issue by MridulS** [networkx/outreachy#3]
[Everyone can do this task. There is no need to claim/assign this issue]

Make sure you have completed the [first task](https://github.com/networkx/outreachy/issues/2) before proceeding further. In this task applicants are expected to setup their local development environment, required to contribute code to the networkx project. Applicants should go through our [Contributor Guide](https://networkx.org/documentation/latest/developer/contribute.html).

Once you have installed all the requirements and setup the environment, please run the [tests locally](https://networkx.org/documentation/latest/developer/contribute.html#testing). If all the tests pass then copy the output of `$ PYTHONPATH=. pytest networkx` and put it inside a file named `nx_dev_test_output.txt`. Add this file to your folder created in the [first task](https://github.com/networkx/outreachy/issues/2) and push the changes to github.

Create a pull request against this repository and add the file  `2022-round-1/<your-github-username>/nx_dev_test_output.txt` with the pytest output.

**Comment by Saumay1** [comment]
Hi, @MridulS!
I am trying to set up build environment using conda but am getting this issue:

![Screenshot (291)](https://user-images.githubusercontent.com/46322717/161786813-09cd0f3e-a97e-4619-a4ce-e19159277ad2.png)
 
CondaFileIOError: 'requirements/default.txt'. [Errno 2] No such file or directory: 'requirements/default.txt'
when running conda install -c conda-forge --file requirements/default.txt --file requirements/test.txt --file requirements/developer.txt

I tried finding the solution but couldn't.
I am using Windows.
Could you please help me with this?

**Comment by PurviChaurasia** [comment]
Hi, I am stuck in creating a virtual env in vscode. The following command is not working - 
`source networkx-dev/bin/activate`
The error is that source is not recognized as an internal or external command. 
Can someone help me out?

**Comment by Ruchi-here** [comment]
Yeah, its because the command given is Linux command and you might be using on Windows

**Comment by PurviChaurasia** [comment]
Oh yes! Thanks for pointing it out :)

**Comment by khushishikhu** [comment]
While I was refering to the Contributor Guide. In the virtual environment step- I ran
 `# Install networkx from source
pip install -e . ` on windows. (my virtualenv is also working)

I showed me "ERROR: Package 'networkx' requires a different Python: 3.8.0 not in '>=3.9'" this error. Can you please help to resolve it
